bitbots_ros_control/scripts/imu_interactive_marker.py

```diff
@@ -0,0 +1,116 @@
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of the Willow Garage, Inc. nor the names of its
+      contributors may be used to endorse or promote products derived from
+      this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES LOSS OF USE, DATA, OR PROFITS OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+"""
+import traceback
+
+import rospy
+import copy
+import math
+
+from interactive_markers.interactive_marker_server import InteractiveMarkerServer
+from visualization_msgs.msg import InteractiveMarker, InteractiveMarkerControl, Marker
+from geometry_msgs.msg import Pose, Point, Quaternion, Vector3
+from sensor_msgs.msg import Imu
+
+rospy.init_node("imu_interactive_marker")
+
+def normalizeQuaternion( quaternion_msg ):
+    norm = quaternion_msg.x**2 + quaternion_msg.y**2 + quaternion_msg.z**2 + quaternion_msg.w**2
+    s = norm**(-0.5)
+    quaternion_msg.x *= s
+    quaternion_msg.y *= s
+    quaternion_msg.z *= s
+    quaternion_msg.w *= s
+
+class IMUMarker:
+
+    def __init__(self, server):
+        self.marker_name = "IMU"
+        self.imu_publisher = rospy.Publisher("/imu/data", Imu, queue_size=1)
+        self.server = server
+        self.pose = Pose()
+        self.pose.orientation.w = 1
+
+        int_marker = InteractiveMarker()
+        int_marker.header.frame_id = "imu_frame"
+        int_marker.pose = self.pose
+        int_marker.scale = 1
+        int_marker.name = self.marker_name
+        int_marker.description = "Rotate 2DOF to simulate IMU orientation to ground"
+
+        control = InteractiveMarkerControl()
+        control.orientation.w = 1
+        control.orientation.x = 1
+        control.orientation.y = 0
+        control.orientation.z = 0
+        normalizeQuaternion(control.orientation)
+        control.name = "rotate_x"
+        control.interaction_mode = InteractiveMarkerControl.ROTATE_AXIS
+        control.always_visible = True
+        int_marker.controls.append(control)
+
+        control = InteractiveMarkerControl()
+        control.orientation.w = 1
+        control.orientation.x = 0
+        control.orientation.y = 0
+        control.orientation.z = 1
+        normalizeQuaternion(control.orientation)
+        control.name = "rotate_y"
```

Comment:
Why is this called `rotate_y` when it rotates around z? Or I understood this wrongly. not really sure,

Reply:
the orientation is a quaternion. it works different than euler angles (roll pitch yaw)

Reply:
It does rotate around y axis thou